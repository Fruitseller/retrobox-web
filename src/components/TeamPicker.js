import React, { Component } from 'react';
import { withRouter } from 'react-router';
import { withStyles } from 'material-ui/styles';
import Typography from 'material-ui/Typography';
import Input from 'material-ui/Input';
import Button from 'material-ui/Button';
import RetroAppBar from './RetroAppBar';

const styles = theme => ({
  input: {
    margin: theme.spacing.unit
  },
  button: {
    margin: theme.spacing.unit
  }
});

class TeamPicker extends Component {
  goToTeam = event => {
    event.preventDefault();
    const teamId = this.teamInput.value.toLowerCase();
    this.props.history.push(`team/${teamId}`);
  };

  render() {
    const { classes, handleAuthentication, authenticationText } = this.props;
    return (
      <div>
        <RetroAppBar
          handleOnClick={handleAuthentication}
          buttonText={authenticationText}
        />
        <form className="team-selector" onSubmit={this.goToTeam}>
          <Typography type="headline" component="h2">
            Please Enter a Teamname
          </Typography>

          <Input
            placeholder="Team Name"
            className={classes.input}
            required
            inputProps={{
              'aria-label': 'Description'
            }}
            inputRef={input => {
              this.teamInput = input;
            }}
          />
          <Button raised type="submit" className={classes.button}>
            Open team Retrobox
          </Button>
        </form>
      </div>
    );
  }
}

export default withRouter(withStyles(styles)(TeamPicker));
