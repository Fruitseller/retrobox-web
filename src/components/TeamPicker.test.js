import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import TeamPicker from './TeamPicker';

configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  shallow(<TeamPicker />);
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Router>
        <TeamPicker />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
