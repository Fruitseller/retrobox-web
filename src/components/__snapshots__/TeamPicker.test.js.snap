// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="team-selector"
  onSubmit={[Function]}
>
  <h2
    className="MuiTypography-root-3 MuiTypography-headline-8"
  >
    Please Enter a Teamname
  </h2>
  <div
    className="MuiInput-root-27 MuiInput-inkbar-29 MuiInput-underline-35 TeamPicker-input-1"
    onBlur={[Function]}
    onFocus={[Function]}
    required={true}
  >
    <input
      aria-label="Description"
      autoComplete={undefined}
      autoFocus={undefined}
      className="MuiInput-input-31 MuiInput-inputSingleline-38"
      defaultValue={undefined}
      disabled={undefined}
      id={undefined}
      name={undefined}
      onChange={[Function]}
      onKeyDown={undefined}
      onKeyUp={undefined}
      placeholder="Team Name"
      readOnly={undefined}
      required={undefined}
      rows={undefined}
      type="text"
      value={undefined}
    />
  </div>
  <button
    className="MuiButtonBase-root-56 MuiButton-root-42 MuiButton-raised-49 TeamPicker-button-2"
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
    type="submit"
  >
    <span
      className="MuiButton-label-44"
    >
      Open team Retrobox
    </span>
    <span
      className="MuiTouchRipple-root-58"
    />
  </button>
</form>
`;
